app/src/hooks/activity-value-form/use-activity-form.ts

```diff
@@ -0,0 +1,78 @@
+import { useForm } from "react-hook-form";
+import { useEffect } from "react";
+import { ActivityValue } from "@/models/ActivityValue";
+import { SuggestedActivity } from "@/util/form-schema";
+import { Inputs } from "@/components/Modals/activity-modal/activity-modal-body";
+
+const useActivityForm = ({
+  targetActivityValue,
+  selectedActivity,
+}: {
+  targetActivityValue: ActivityValue | undefined;
+  selectedActivity?: SuggestedActivity;
+}) => {
+  const {
+    register,
+    handleSubmit,
+    reset,
+    watch,
+    setError,
+    setFocus,
+    formState: { errors },
+  } = useForm<Inputs>();
+
+  useEffect(() => {
+    if (targetActivityValue) {
+      reset({
+        activity: {
+          ...targetActivityValue.activityData,
+          dataQuality: targetActivityValue?.dataSource?.dataQuality,
+          sourceReference: targetActivityValue?.dataSource?.notes,
+          CH4EmissionFactor: targetActivityValue?.activityData?.ch4_amount,
+          CO2EmissionFactor: targetActivityValue?.activityData?.co2_amount,
+          N2OEmissionFactor: targetActivityValue?.activityData?.n2o_amount,
+          activityDataAmount: 0,
+          activityDataUnit: null,
+          emissionFactorType: "",
+          totalFuelConsumption: "",
+          totalFuelConsumptionUnits: "",
+          co2EmissionFactorUnit: "",
+          n2oEmissionFactorUnit: "",
+          ch4EmissionFactorUnit: "",
+        },
+      });
+    } else {
+      reset({
+        activity: {
+          activityType: selectedActivity?.id,
+          fuelType: "",
+          dataQuality: "",
+          sourceReference: "",
+          CH4EmissionFactor: 0,
+          CO2EmissionFactor: 0,
+          N2OEmissionFactor: 0,
+          activityDataAmount: 0,
+          activityDataUnit: null,
+          emissionFactorType: "",
+          totalFuelConsumption: "",
+          totalFuelConsumptionUnits: "",
+          co2EmissionFactorUnit: "",
+          n2oEmissionFactorUnit: "",
+          ch4EmissionFactorUnit: "",
+        },
```

Comment:
If you store this default value object for the form next to the type definition, you can use is above as well to set the defaults for things you're not overwriting.